torch/csrc/autograd/variable.h

```diff
@@ -511,15 +468,21 @@ struct TORCH_API Variable::Impl : public at::TensorImpl {
 /// through the view relation.
 /// Relevant logic for non-differentiable views is implemented in
 /// make_variable_view below, and wrap_output of gen_variable_type.py.
-struct TORCH_API Variable::DifferentiableViewImpl : public Variable::Impl {
-  DifferentiableViewImpl(
-    Variable base,
-    at::Tensor data,
-    Edge gradient_edge,
-    std::unique_ptr<Variable::DifferentiableViewMeta> autograd_meta);
+struct TORCH_API Variable::DifferentiableViewMeta : public Variable::AutogradMeta {
+  /// The base `Variable` (never a view).
+  Variable base_;
+
+  /// The value of the version_counter at the time grad_fn was created. The
+  /// grad_fn field is stale if attr_version !=
+  /// version_counter.current_version().
+  uint32_t attr_version;
 
-  /// Reset all expensive fields to free up resources
-  void release_resources() override;
+  bool requires_grad() const override {
+    return requires_grad_ || grad_fn_ || (is_view_ && base_.requires_grad());
+  }
+
+  DifferentiableViewMeta(at::TensorImpl* self_impl, Variable base, Edge gradient_edge);
+  ~DifferentiableViewMeta();
```

Comment:
should this be virtual?

Reply:
I think we don't need to mark this as `virtual`, since `DifferentiableViewMeta` subclasses from `AutogradMeta`, which subclasses from `c10::AutogradMetaInterface`, and `c10::AutogradMetaInterface`'s destructor (https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.h#L139) is already `virtual`. Based on https://stackoverflow.com/a/5610164, `~DifferentiableViewMeta()` will be `virtual` automatically as well.